openedx/core/djangoapps/content/search/documents.py

```diff
@@ -223,6 +227,35 @@ def _tags_for_content_object(object_id: UsageKey | LearningContextKey) -> dict:
     return {Fields.tags: result}
 
 
+def _collections_for_content_object(object_id: UsageKey | LearningContextKey) -> dict:
+    """
+    Given an XBlock, course, library, etc., get the collections for its index doc.
+
+    e.g. for something in Collections "COL_A" and "COL_B", this would return:
+        {
+            "collections": ["COL_A", "COL_B"],
```

Comment:
@pomegranited Sorry for the late request (I overlooked this before), but what about changing the structure here?

```suggestion
            "collections": [
                { "display_name": "Collection A", key: "COL_A" },
                { "display_name": "Collection B", key: "COL_B" },
            ],
```

That way, we can use the `display_name` as a searchable attribute and the `slug/key` to a dev action (like a redirect).

Reply:
Sure @rpenido , but I decided to store the collections data like we store the tags data, e.g.

```
   "collections": {
      "display_name": ["Collection One", "Another Collection"],
      "key": ["collection-one", "another-key"],
   },
```

cf a81ea9afd20a27918f376983f65cb7ee1c32667c